### ISSUE TITLE
Chocolatey's 'cup all' seems to be installing 32bit packages on my 64bit machine

### ISSUE THREAD

**Issue by fabiorzfreitas** [chocolatey/choco#2795]
I'm on Windows 10 21H2 x64, using choco 1.1.0.

It seems that almost every time I run `cup all -y`, Chocolatey tries to retrieve and install 32bit versions of packages regardless if I already have their 64bit versions installed.

I'll paste below both the relevant output of `cup all -yvd` for the package `googledrive`, as well as for `cup googledrive -yvdf` (the latter install ended failing for other reasons, but the installer that was ran was x64). Upon closer inspection, the `cup all -yvd` seems to show that the user added `-forceX86`, which I did not:

<details>
    <summary><code>cup all -yvd</code> output for <code>googledrive</code>:</summary>

```
googledrive v62.0.1.0 [Approved]
googledrive package files upgrade completed. Performing other installation steps.
Setting installer args for googledrive
Setting package parameters for googledrive
Contents of 'C:\ProgramData\chocolatey\lib\googledrive\tools\chocolateyInstall.ps1':
$ErrorActionPreference = 'Stop'

$toolsDir     = "$(Split-Path -parent $MyInvocation.MyCommand.Definition)"
$url          = 'https://dl.google.com/drive-file-stream/GoogleDriveSetup.exe'
$checksum     = '631A1C4C9024CE17D446CDC47A64D61E084368B233AF6309CC8833CF046F518E'
$checksumType = 'sha256'

$packageArgs = @{
  packageName    = $env:ChocolateyPackageName
  unzipLocation  = $toolsDir
  fileType       = 'exe'
  url            = $url
  checksum       = $checksum
  checksumType   = $checksumType
  softwareName  = 'Google Drive*'
  silentArgs    = "--silent --desktop_shortcut"
  validExitCodes= @(0,1641,3010)
}

Install-ChocolateyPackage @packageArgs


Calling built-in PowerShell host with ['[System.Threading.Thread]::CurrentThread.CurrentCulture = '';[System.Threading.Thread]::CurrentThread.CurrentUICulture = ''; & import-module -name 'C:\ProgramData\chocolatey\helpers\chocolateyInstaller.psm1'; & 'C:\ProgramData\chocolatey\helpers\chocolateyScriptRunner.ps1' -packageScript 'C:\ProgramData\chocolatey\lib\googledrive\tools\chocolateyInstall.ps1' -installArguments 'ADD_CMAKE_TO_PATH=System' -packageParameters '' -forceX86']
Host version is 5.1.19041.1, PowerShell Version is '5.1.19041.1682' and CLR Version is '4.0.30319.42000'.
Loading community extensions
Importing 'C:\ProgramData\chocolatey\extensions\chocolatey-compatibility\chocolatey-compatibility.psm1'
Function 'Get-PackageParameters' exists, ignoring export.
Function 'Get-UninstallRegistryKey' exists, ignoring export.
Exporting function 'Install-ChocolateyDesktopLink' for backwards compatibility
Exporting function 'Write-ChocolateyFailure' for backwards compatibility
Exporting function 'Write-ChocolateySuccess' for backwards compatibility
Exporting function 'Write-FileUpdateLog' for backwards compatibility
Importing 'C:\ProgramData\chocolatey\extensions\chocolatey-core\chocolatey-core.psm1'
Importing 'C:\ProgramData\chocolatey\extensions\chocolatey-dotnetfx\chocolatey-dotnetfx.psm1'
Importing 'C:\ProgramData\chocolatey\extensions\chocolatey-fastanswers\chocolateyinstall-helpers.psm1'
Importing 'C:\ProgramData\chocolatey\extensions\chocolatey-misc-helpers\chocolatey-misc-helpers.psm1'
Importing 'C:\ProgramData\chocolatey\extensions\chocolatey-visualstudio\chocolatey-visualstudio.extension.psm1'
Importing 'C:\ProgramData\chocolatey\extensions\chocolatey-windowsupdate\chocolatey-windowsupdate.psm1'
---------------------------Script Execution---------------------------
Running 'ChocolateyScriptRunner' for googledrive v62.0.1.0 with packageScript 'C:\ProgramData\chocolatey\lib\googledrive\tools\chocolateyInstall.ps1', packageFolder:'C:\ProgramData\chocolatey\lib\googledrive', installArguments: 'ADD_CMAKE_TO_PATH=System', packageParameters: '',
Running 'C:\ProgramData\chocolatey\lib\googledrive\tools\chocolateyInstall.ps1'
Running Install-ChocolateyPackage -url 'https://dl.google.com/drive-file-stream/GoogleDriveSetup.exe' -checksum '631A1C4C9024CE17D446CDC47A64D61E084368B233AF6309CC8833CF046F518E' -packageName 'googledrive' -checksumType 'sha256' -fileType 'exe' -validExitCodes '0 1641 3010' -silentArgs '--silent --desktop_shortcut'
Running Get-ChocolateyWebFile -packageName 'googledrive' -fileFullPath 'C:\Users\Fabio\AppData\Local\Temp\chocolatey\googledrive\62.0.1.0\googledriveInstall.exe' -url 'https://dl.google.com/drive-file-stream/GoogleDriveSetup.exe' -url64bit '' -checksum '631A1C4C9024CE17D446CDC47A64D61E084368B233AF6309CC8833CF046F518E' -checksumType 'sha256' -checksum64 '' -checksumType64 '' -options 'System.Collections.Hashtable' -getOriginalFileName 'True'
Running Get-ProcessorBits -compare '64'
CPU is 64 bit
User specified -x86 so forcing 32 bit
Running Get-WebFileName -url 'https://dl.google.com/drive-file-stream/GoogleDriveSetup.exe' -defaultName 'googledriveInstall.exe'
Using response url to determine file name. 'https://dl.google.com/drive-file-stream/GoogleDriveSetup.exe'
File name determined from url is 'GoogleDriveSetup.exe'
Running Get-WebHeaders -url 'https://dl.google.com/drive-file-stream/GoogleDriveSetup.exe' -ErrorAction 'Stop'
Setting the UserAgent to 'chocolatey command line'
Request Headers:
  'Accept':'*/*'
  'User-Agent':'chocolatey command line'
Response Headers:
  'Content-Disposition':'attachment'
  'Content-Security-Policy':'default-src 'none''
  'X-Content-Type-Options':'nosniff'
  'X-Frame-Options':'SAMEORIGIN'
  'X-Xss-Protection':'0'
  'Age':'0'
  'Alt-Svc':'h3=":443"; ma=2592000,h3-29=":443"; ma=2592000,h3-Q050=":443"; ma=2592000,h3-Q046=":443"; ma=2592000,h3-Q043=":443"; ma=2592000,quic=":443"; ma=2592000; v="46,43"'
  'Accept-Ranges':'bytes'
  'Content-Length':'285304136'
  'Cache-Control':'public,max-age=0'
  'Content-Type':'application/octet-stream'
  'Date':'Tue, 16 Aug 2022 17:32:03 GMT'
  'ETag':'"ed54b4"'
  'Last-Modified':'Thu, 28 Jul 2022 19:08:10 GMT'
  'Server':'downloads'
Downloading googledrive 32 bit
  from 'https://dl.google.com/drive-file-stream/GoogleDriveSetup.exe'
Running Get-WebFile -url 'https://dl.google.com/drive-file-stream/GoogleDriveSetup.exe' -fileName 'C:\Users\Fabio\AppData\Local\Temp\chocolatey\googledrive\62.0.1.0\GoogleDriveSetup.exe' -options 'System.Collections.Hashtable'
Setting request timeout to  30000
Setting read/write timeout to  2700000
Setting the UserAgent to 'chocolatey command line'
Downloading https://dl.google.com/drive-file-stream/GoogleDriveSetup.exe to C:\Users\Fabio\AppData\Local\Temp\chocolatey\googledrive\62.0.1.0\GoogleDriveSetup.exe
Progress: 23% - Saving 63.34 MB of 272.09 MB
```

</details>

<details>
    <summary><code>cup googledrive -yvdf</code> output:</summary>

```
Chocolatey v1.1.0
Chocolatey is running on Windows v 10.0.19044.0
Attempting to delete file "C:/ProgramData/chocolatey/choco.exe.old".
Attempting to delete file "C:\ProgramData\chocolatey\choco.exe.old".
Command line: "C:\ProgramData\chocolatey\choco.exe" upgrade googledrive -yvdf
Received arguments: upgrade googledrive -yvdf
RemovePendingPackagesTask is now ready and waiting for PreRunMessage.
Sending message 'PreRunMessage' out if there are subscribers...
[Pending] Removing all pending packages that should not be considered installed...
Performing validation checks.
Global Configuration Validation Checks:
 - Package Exit Code / Exit On Reboot = Checked
System State Validation Checks:
 Reboot Requirement Checks:
 - Pending Computer Rename = Checked
 - Pending Component Based Servicing = Checked
 - Pending Windows Auto Update = Checked
 - Pending File Rename Operations = Ignored
 - Pending Windows Package Installer = Checked
 - Pending Windows Package Installer SysWow64 = Checked
The source 'https://chocolatey.org/api/v2/' evaluated to a 'normal' source type

NOTE: Hiding sensitive configuration data! Please double and triple
 check to be sure no sensitive data is shown, especially if copying
 output to a gist for review.
Configuration: CommandName='upgrade'|
CacheLocation='C:\Users\Fabio\AppData\Local\Temp\chocolatey'|
ContainsLegacyPackageInstalls='True'|
CommandExecutionTimeoutSeconds='2700'|WebRequestTimeoutSeconds='30'|
Sources='https://chocolatey.org/api/v2/'|SourceType='normal'|
Debug='True'|Verbose='True'|Trace='False'|Force='True'|Noop='False'|
HelpRequested='False'|UnsuccessfulParsing='False'|RegularOutput='True'|
QuietOutput='False'|PromptForConfirmation='False'|
DisableCompatibilityChecks='False'|AcceptLicense='True'|
AllowUnofficialBuild='False'|Input='googledrive'|AllVersions='False'|
SkipPackageInstallProvider='False'|PackageNames='googledrive'|
Prerelease='False'|ForceX86='False'|OverrideArguments='False'|
NotSilent='False'|ApplyPackageParametersToDependencies='False'|
ApplyInstallArgumentsToDependencies='False'|IgnoreDependencies='False'|
AllowMultipleVersions='False'|AllowDowngrade='False'|
ForceDependencies='False'|Information.PlatformType='Windows'|
Information.PlatformVersion='10.0.19044.0'|
Information.PlatformName='Windows 10'|
Information.ChocolateyVersion='1.1.0.0'|
Information.ChocolateyProductVersion='1.1.0'|
Information.FullName='choco, Version=1.1.0.0, Culture=neutral, PublicKeyToken=79d02ea9cad655eb'|

Information.Is64BitOperatingSystem='True'|
Information.Is64BitProcess='True'|Information.IsInteractive='True'|
Information.UserName='Fabio'|Information.UserDomainName='FABIO-PC'|
Information.IsUserAdministrator='True'|
Information.IsUserSystemAccount='False'|
Information.IsUserRemoteDesktop='False'|
Information.IsUserRemote='True'|
Information.IsProcessElevated='True'|
Information.IsLicensedVersion='False'|Information.LicenseType='Foss'|
Information.CurrentDirectory='C:\WINDOWS\system32'|
Features.AutoUninstaller='True'|Features.ChecksumFiles='True'|
Features.AllowEmptyChecksums='False'|
Features.AllowEmptyChecksumsSecure='True'|
Features.FailOnAutoUninstaller='False'|
Features.FailOnStandardError='False'|Features.UsePowerShellHost='True'|
Features.LogEnvironmentValues='False'|Features.LogWithoutColor='False'|
Features.VirusCheck='False'|
Features.FailOnInvalidOrMissingLicense='False'|
Features.IgnoreInvalidOptionsSwitches='True'|
Features.UsePackageExitCodes='True'|
Features.UseEnhancedExitCodes='False'|
Features.UseFipsCompliantChecksums='False'|
Features.ShowNonElevatedWarnings='True'|
Features.ShowDownloadProgress='True'|
Features.StopOnFirstPackageFailure='False'|
Features.UseRememberedArgumentsForUpgrades='True'|
Features.IgnoreUnfoundPackagesOnUpgradeOutdated='False'|
Features.SkipPackageUpgradesWhenNotInstalled='False'|
Features.RemovePackageInformationOnUninstall='False'|
Features.ExitOnRebootDetected='False'|
Features.LogValidationResultsOnWarnings='True'|
Features.UsePackageRepositoryOptimizations='True'|
ListCommand.LocalOnly='False'|ListCommand.IdOnly='False'|
ListCommand.IncludeRegistryPrograms='False'|ListCommand.PageSize='25'|
ListCommand.Exact='False'|ListCommand.ByIdOnly='False'|
ListCommand.ByTagOnly='False'|ListCommand.IdStartsWith='False'|
ListCommand.OrderByPopularity='False'|ListCommand.ApprovedOnly='False'|
ListCommand.DownloadCacheAvailable='False'|
ListCommand.NotBroken='False'|
ListCommand.IncludeVersionOverrides='False'|
UpgradeCommand.FailOnUnfound='False'|
UpgradeCommand.FailOnNotInstalled='False'|
UpgradeCommand.NotifyOnlyAvailableUpgrades='False'|
UpgradeCommand.ExcludePrerelease='False'|
NewCommand.AutomaticPackage='False'|
NewCommand.UseOriginalTemplate='False'|SourceCommand.Command='unknown'|
SourceCommand.Priority='0'|SourceCommand.BypassProxy='False'|
SourceCommand.AllowSelfService='False'|
SourceCommand.VisibleToAdminsOnly='False'|
FeatureCommand.Command='unknown'|ConfigCommand.Command='unknown'|
ApiKeyCommand.Remove='False'|PinCommand.Command='unknown'|
OutdatedCommand.IgnorePinned='False'|
ExportCommand.IncludeVersionNumbers='False'|Proxy.BypassOnLocal='True'|
TemplateCommand.Command='unknown'|
_ Chocolatey:ChocolateyUpgradeCommand - Normal Run Mode _
Upgrading the following packages:
googledrive
By upgrading, you accept licenses for the packages.
Using 'https://chocolatey.org/api/v2/'.
- Supports prereleases? 'True'.
- Is ServiceBased? 'True'.
Package 'googledrive' found on source 'https://chocolatey.org/api/v2/'
googledrive v62.0.1.0 is the latest version available based on your source(s).
Attempting to delete directory "C:\ProgramData\chocolatey\lib-bkp\googledrive".
Backing up existing googledrive prior to operation.
Moving 'C:\ProgramData\chocolatey\lib\googledrive'
 to 'C:\ProgramData\chocolatey\lib-bkp\googledrive'
Attempting to create directory "C:\ProgramData\chocolatey\lib\googledrive".
Attempting to copy "C:\ProgramData\chocolatey\lib-bkp\googledrive\.chocolateyPending"
 to "C:\ProgramData\chocolatey\lib\googledrive\.chocolateyPending".
Attempting to copy "C:\ProgramData\chocolatey\lib-bkp\googledrive\googledrive.nupkg"
 to "C:\ProgramData\chocolatey\lib\googledrive\googledrive.nupkg".
Attempting to copy "C:\ProgramData\chocolatey\lib-bkp\googledrive\googledrive.nuspec"
 to "C:\ProgramData\chocolatey\lib\googledrive\googledrive.nuspec".
Attempting to create directory "C:\ProgramData\chocolatey\lib\googledrive\tools".
Attempting to copy "C:\ProgramData\chocolatey\lib-bkp\googledrive\tools\chocolateyInstall.ps1"
 to "C:\ProgramData\chocolatey\lib\googledrive\tools\chocolateyInstall.ps1".
Deleting file C:\ProgramData\chocolatey\lib\googledrive\tools\chocolateyInstall.ps1
Attempting to delete file "C:\ProgramData\chocolatey\lib\googledrive\tools\chocolateyInstall.ps1".
Attempting to delete directory "C:\ProgramData\chocolatey\lib\googledrive".
Ensuring removal of package cache files.
Attempting to delete directory "C:\Users\Fabio\AppData\Local\Temp\chocolatey\googledrive\62.0.1.0".
[NuGet] Installing 'googledrive 62.0.1.0'.
[NuGet] Added file 'chocolateyInstall.ps1' to folder 'googledrive\tools'.
[NuGet] Added file 'googledrive.nupkg' to folder 'googledrive'.
[NuGet] Added file 'googledrive.nuspec' to folder 'googledrive'.
[NuGet] Successfully installed 'googledrive 62.0.1.0'.

googledrive v62.0.1.0 (forced) [Approved]
googledrive package files upgrade completed. Performing other installation steps.
Setting installer args for googledrive
Setting package parameters for googledrive
Contents of 'C:\ProgramData\chocolatey\lib\googledrive\tools\chocolateyInstall.ps1':
$ErrorActionPreference = 'Stop'

$toolsDir     = "$(Split-Path -parent $MyInvocation.MyCommand.Definition)"
$url          = 'https://dl.google.com/drive-file-stream/GoogleDriveSetup.exe'
$checksum     = '631A1C4C9024CE17D446CDC47A64D61E084368B233AF6309CC8833CF046F518E'
$checksumType = 'sha256'

$packageArgs = @{
  packageName    = $env:ChocolateyPackageName
  unzipLocation  = $toolsDir
  fileType       = 'exe'
  url            = $url
  checksum       = $checksum
  checksumType   = $checksumType
  softwareName  = 'Google Drive*'
  silentArgs    = "--silent --desktop_shortcut"
  validExitCodes= @(0,1641,3010)
}

Install-ChocolateyPackage @packageArgs


Calling built-in PowerShell host with ['[System.Threading.Thread]::CurrentThread.CurrentCulture = '';[System.Threading.Thread]::CurrentThread.CurrentUICulture = ''; & import-module -name 'C:\ProgramData\chocolatey\helpers\chocolateyInstaller.psm1'; & 'C:\ProgramData\chocolatey\helpers\chocolateyScriptRunner.ps1' -packageScript 'C:\ProgramData\chocolatey\lib\googledrive\tools\chocolateyInstall.ps1' -installArguments '' -packageParameters ''']
Host version is 5.1.19041.1, PowerShell Version is '5.1.19041.1682' and CLR Version is '4.0.30319.42000'.
Loading community extensions
Importing 'C:\ProgramData\chocolatey\extensions\chocolatey-compatibility\chocolatey-compatibility.psm1'
Function 'Get-PackageParameters' exists, ignoring export.
Function 'Get-UninstallRegistryKey' exists, ignoring export.
Exporting function 'Install-ChocolateyDesktopLink' for backwards compatibility
Exporting function 'Write-ChocolateyFailure' for backwards compatibility
Exporting function 'Write-ChocolateySuccess' for backwards compatibility
Exporting function 'Write-FileUpdateLog' for backwards compatibility
Importing 'C:\ProgramData\chocolatey\extensions\chocolatey-core\chocolatey-core.psm1'
Importing 'C:\ProgramData\chocolatey\extensions\chocolatey-dotnetfx\chocolatey-dotnetfx.psm1'
Importing 'C:\ProgramData\chocolatey\extensions\chocolatey-fastanswers\chocolateyinstall-helpers.psm1'
Importing 'C:\ProgramData\chocolatey\extensions\chocolatey-misc-helpers\chocolatey-misc-helpers.psm1'
Importing 'C:\ProgramData\chocolatey\extensions\chocolatey-visualstudio\chocolatey-visualstudio.extension.psm1'
Importing 'C:\ProgramData\chocolatey\extensions\chocolatey-windowsupdate\chocolatey-windowsupdate.psm1'
---------------------------Script Execution---------------------------
Running 'ChocolateyScriptRunner' for googledrive v62.0.1.0 with packageScript 'C:\ProgramData\chocolatey\lib\googledrive\tools\chocolateyInstall.ps1', packageFolder:'C:\ProgramData\chocolatey\lib\googledrive', installArguments: '', packageParameters: '',
Running 'C:\ProgramData\chocolatey\lib\googledrive\tools\chocolateyInstall.ps1'
Running Install-ChocolateyPackage -url 'https://dl.google.com/drive-file-stream/GoogleDriveSetup.exe' -checksum '631A1C4C9024CE17D446CDC47A64D61E084368B233AF6309CC8833CF046F518E' -packageName 'googledrive' -checksumType 'sha256' -fileType 'exe' -validExitCodes '0 1641 3010' -silentArgs '--silent --desktop_shortcut'
Running Get-ChocolateyWebFile -packageName 'googledrive' -fileFullPath 'C:\Users\Fabio\AppData\Local\Temp\chocolatey\googledrive\62.0.1.0\googledriveInstall.exe' -url 'https://dl.google.com/drive-file-stream/GoogleDriveSetup.exe' -url64bit '' -checksum '631A1C4C9024CE17D446CDC47A64D61E084368B233AF6309CC8833CF046F518E' -checksumType 'sha256' -checksum64 '' -checksumType64 '' -options 'System.Collections.Hashtable' -getOriginalFileName 'True'
Running Get-ProcessorBits -compare '64'
CPU is 64 bit
Running Get-WebFileName -url 'https://dl.google.com/drive-file-stream/GoogleDriveSetup.exe' -defaultName 'googledriveInstall.exe'
Using response url to determine file name. 'https://dl.google.com/drive-file-stream/GoogleDriveSetup.exe'
File name determined from url is 'GoogleDriveSetup.exe'
Running Get-WebHeaders -url 'https://dl.google.com/drive-file-stream/GoogleDriveSetup.exe' -ErrorAction 'Stop'
Setting the UserAgent to 'chocolatey command line'
Request Headers:
  'Accept':'*/*'
  'User-Agent':'chocolatey command line'
Response Headers:
  'Content-Disposition':'attachment'
  'Content-Security-Policy':'default-src 'none''
  'X-Content-Type-Options':'nosniff'
  'X-Frame-Options':'SAMEORIGIN'
  'X-Xss-Protection':'0'
  'Age':'0'
  'Alt-Svc':'h3=":443"; ma=2592000,h3-29=":443"; ma=2592000,h3-Q050=":443"; ma=2592000,h3-Q046=":443"; ma=2592000,h3-Q043=":443"; ma=2592000,quic=":443"; ma=2592000; v="46,43"'
  'Accept-Ranges':'bytes'
  'Content-Length':'285304136'
  'Cache-Control':'public,max-age=0'
  'Content-Type':'application/octet-stream'
  'Date':'Tue, 16 Aug 2022 17:42:42 GMT'
  'ETag':'"ed54b4"'
  'Last-Modified':'Thu, 28 Jul 2022 19:08:10 GMT'
  'Server':'downloads'
Downloading googledrive
  from 'https://dl.google.com/drive-file-stream/GoogleDriveSetup.exe'
Running Get-WebFile -url 'https://dl.google.com/drive-file-stream/GoogleDriveSetup.exe' -fileName 'C:\Users\Fabio\AppData\Local\Temp\chocolatey\googledrive\62.0.1.0\GoogleDriveSetup.exe' -options 'System.Collections.Hashtable'
Setting request timeout to  30000
Setting read/write timeout to  2700000
Setting the UserAgent to 'chocolatey command line'
Downloading https://dl.google.com/drive-file-stream/GoogleDriveSetup.exe to C:\Users\Fabio\AppData\Local\Temp\chocolatey\googledrive\62.0.1.0\GoogleDriveSetup.exe
Progress: 100% - Completed download of C:\Users\Fabio\AppData\Local\Temp\chocolatey\googledrive\62.0.1.0\GoogleDriveSetup.exe (272.09 MB).
Download of GoogleDriveSetup.exe (272.09 MB) completed.
No runtime virus checking built into FOSS Chocolatey. Check out Pro/Business - https://chocolatey.org/compare
Verifying package provided checksum of '631A1C4C9024CE17D446CDC47A64D61E084368B233AF6309CC8833CF046F518E' for 'C:\Users\Fabio\AppData\Local\Temp\chocolatey\googledrive\62.0.1.0\GoogleDriveSetup.exe'.
Running Get-ChecksumValid -file 'C:\Users\Fabio\AppData\Local\Temp\chocolatey\googledrive\62.0.1.0\GoogleDriveSetup.exe' -checksum '631A1C4C9024CE17D446CDC47A64D61E084368B233AF6309CC8833CF046F518E' -checksumType 'sha256' -originalUrl 'https://dl.google.com/drive-file-stream/GoogleDriveSetup.exe'
checksum.exe found at 'C:\ProgramData\chocolatey\helpers\..\tools\checksum.exe'
Executing command ['C:\ProgramData\chocolatey\helpers\..\tools\checksum.exe' -c="631A1C4C9024CE17D446CDC47A64D61E084368B233AF6309CC8833CF046F518E" -t="sha256" -f="C:\Users\Fabio\AppData\Local\Temp\chocolatey\googledrive\62.0.1.0\GoogleDriveSetup.exe"]
Hashes match.
Command ['C:\ProgramData\chocolatey\helpers\..\tools\checksum.exe' -c="631A1C4C9024CE17D446CDC47A64D61E084368B233AF6309CC8833CF046F518E" -t="sha256" -f="C:\Users\Fabio\AppData\Local\Temp\chocolatey\googledrive\62.0.1.0\GoogleDriveSetup.exe"] exited with '0'.
Running Install-ChocolateyInstallPackage -packageName 'googledrive' -fileType 'exe' -silentArgs '--silent --desktop_shortcut' -file 'C:\Users\Fabio\AppData\Local\Temp\chocolatey\googledrive\62.0.1.0\GoogleDriveSetup.exe' -validExitCodes '0 1641 3010' -useOnlyPackageSilentArguments 'False'
Running Get-ProcessorBits -compare '32'
Installing googledrive...
Running Start-ChocolateyProcessAsAdmin -validExitCodes '0 1641 3010' -workingDirectory 'C:\Users\Fabio\AppData\Local\Temp\chocolatey\googledrive\62.0.1.0' -statements '--silent --desktop_shortcut ' -exeToRun 'C:\Users\Fabio\AppData\Local\Temp\chocolatey\googledrive\62.0.1.0\GoogleDriveSetup.exe'
Test-ProcessAdminRights: returning True
Elevating permissions and running ["C:\Users\Fabio\AppData\Local\Temp\chocolatey\googledrive\62.0.1.0\GoogleDriveSetup.exe" --silent --desktop_shortcut ]. This may take a while, depending on the statements.
2022-08-16T17:43:14.841ZI [16520:NonCelloThread] i18n.cc:311:InitializeI18n I18n initialized with pt-br
Command ["C:\Users\Fabio\AppData\Local\Temp\chocolatey\googledrive\62.0.1.0\GoogleDriveSetup.exe" --silent --desktop_shortcut ] exited with '1638'.
WARNING: This MSI requires uninstall prior to installing a different version. Please ask the package maintainer(s) to add a check in the chocolateyInstall.ps1 script and uninstall if the software is installed. This is most likely an issue with the 'googledrive' package and not with Chocolatey itself. Please follow up with the package maintainer(s) directly.
ERROR: Running ["C:\Users\Fabio\AppData\Local\Temp\chocolatey\googledrive\62.0.1.0\GoogleDriveSetup.exe" --silent --desktop_shortcut ] was not successful. Exit code was '1638'. Exit code indicates the following: This MSI requires uninstall prior to installing a different version. Please ask the package maintainer(s) to add a check in the chocolateyInstall.ps1 script and uninstall if the software is installed. This is most likely an issue with the 'googledrive' package and not with Chocolatey itself. Please follow up with the package maintainer(s) directly..
 at Start-ChocolateyProcessAsAdmin, C:\ProgramData\chocolatey\helpers\functions\Start-ChocolateyProcessAsAdmin.ps1: line 326
at Install-ChocolateyInstallPackage, C:\ProgramData\chocolatey\helpers\functions\Install-ChocolateyInstallPackage.ps1: line 359
at Install-ChocolateyPackage, C:\ProgramData\chocolatey\helpers\functions\Install-ChocolateyPackage.ps1: line 418
at <ScriptBlock>, C:\ProgramData\chocolatey\lib\googledrive\tools\chocolateyInstall.ps1: line 20
at <ScriptBlock>, C:\ProgramData\chocolatey\helpers\chocolateyScriptRunner.ps1: line 49
at <ScriptBlock>, <No file>: line 1
Built-in PowerShell host called with ['[System.Threading.Thread]::CurrentThread.CurrentCulture = '';[System.Threading.Thread]::CurrentThread.CurrentUICulture = ''; & import-module -name 'C:\ProgramData\chocolatey\helpers\chocolateyInstaller.psm1'; & 'C:\ProgramData\chocolatey\helpers\chocolateyScriptRunner.ps1' -packageScript 'C:\ProgramData\chocolatey\lib\googledrive\tools\chocolateyInstall.ps1' -installArguments '' -packageParameters '''] exited with '1638'.
Calling command ['"C:\WINDOWS\System32\shutdown.exe" /a']
Command ['"C:\WINDOWS\System32\shutdown.exe" /a'] exited with '1116'
Capturing package files in 'C:\ProgramData\chocolatey\lib\googledrive'
 Found 'C:\ProgramData\chocolatey\lib\googledrive\googledrive.nupkg'
  with checksum '2D62976BE128595D1DDBEFF61A3A6CB1'
 Found 'C:\ProgramData\chocolatey\lib\googledrive\googledrive.nuspec'
  with checksum '22575AFC23C421860DBB17FD156E4753'
 Found 'C:\ProgramData\chocolatey\lib\googledrive\tools\chocolateyInstall.ps1'
  with checksum '19B83471340365597C54F5885442A0BD'
Attempting to create directory "C:\ProgramData\chocolatey\.chocolatey\googledrive.62.0.1.0".
There was no original file at 'C:\ProgramData\chocolatey\.chocolatey\googledrive.62.0.1.0\.files'
Attempting to delete file "C:\ProgramData\chocolatey\.chocolatey\googledrive.62.0.1.0\.arguments".
Attempting to delete file "C:\ProgramData\chocolatey\.chocolatey\googledrive.62.0.1.0\.extra".
Attempting to delete file "C:\ProgramData\chocolatey\.chocolatey\googledrive.62.0.1.0\.version".
Attempting to delete file "C:\ProgramData\chocolatey\.chocolatey\googledrive.62.0.1.0\.sxs".
Attempting to delete file "C:\ProgramData\chocolatey\.chocolatey\googledrive.62.0.1.0\.pin".
Attempting to delete directory "C:\ProgramData\chocolatey\lib-bad\googledrive".
Sending message 'HandlePackageResultCompletedMessage' out if there are subscribers...
The upgrade of googledrive was NOT successful.
Error while running 'C:\ProgramData\chocolatey\lib\googledrive\tools\chocolateyInstall.ps1'.
 See log for details.
Moving 'C:\ProgramData\chocolatey\lib\googledrive'
 to 'C:\ProgramData\chocolatey\lib-bad\googledrive'
Attempting rollback
Moving 'C:\ProgramData\chocolatey\lib-bkp\googledrive'
 to 'C:\ProgramData\chocolatey\lib\googledrive'
Attempting to delete file "C:\Users\Fabio\AppData\Local\NuGet\Cache\googledrive.62.0.1.0.nupkg".

Chocolatey upgraded 0/1 packages. 1 packages failed.
 See the log for details (C:\ProgramData\chocolatey\logs\chocolatey.log).

Failures
 - googledrive (exited 1638) - Error while running 'C:\ProgramData\chocolatey\lib\googledrive\tools\chocolateyInstall.ps1'.
 See log for details.
Sending message 'PostRunMessage' out if there are subscribers...
Exiting with 1638
```

</details>

PS: `-vd` was suggested on discord, but I ended up removing all verbose messages due to maximum character limit.

**Comment by pauby** [comment]
@fabiorzfreitas Before we can look at this, can you go back and fully complete the issue template?

**Comment by pauby** [comment]
As an aside. The `googledrie` package doesn't have a specific 64 bit installer. It only has one installer. That may be 32bit or it may be a 32 bit / 64 bit in the one installer. Regardless, Chocolatey CLI isn't selecting a 64bit / 32bit installer to run as there isn't one.

**Comment by gep13** [comment]
@fabiorzfreitas can you also confirm whether or not you have the `useRememberedArgumentsForUpgrades` feature enabled?

**Comment by fabiorzfreitas** [comment]
@pauby  for some reason, the issue appeared with no template when I started writing it. I found it odd, actually, but perhaps this happened due to GitHub prompting me to sign in!

I would proceed to reproduce this behavior with another package, but @gep13 's suggestion already seem to have solved the issue (although even if the feature were on, I don't remember ever passing -x86 to any package).
